src/compiler/utilities.ts

```diff
@@ -123,7 +123,7 @@ namespace ts {
     /* @internal */
     export function hasChangesInResolutions<T>(names: string[], newResolutions: T[], oldResolutions: Map<T>, comparer: (oldResolution: T, newResolution: T) => boolean): boolean {
         if (names.length !== newResolutions.length) {
-            return false;
+            return true;

```

Comment:
`hasChangesInResolutions` is called on resolutions obtained from `resolveModuleNamesReusingOldState` and `resolveTypeReferenceDirectiveNamesWorker`.
Both guarantee in the implementation that the array they return is the same length as `names`.

Changing to an assert.